x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/hooks/use_latest_vulnerabilities.tsx

```diff
@@ -68,6 +105,10 @@ export const useLatestVulnerabilities = (options: VulnerabilitiesQuery) => {
       keepPreviousData: true,
       enabled: options.enabled,
       onError: (err: Error) => showErrorToast(toasts, err),
+      getNextPageParam: (lastPage) => {
```

Comment:
I realized that I don't fully understand the usage of `useInfiniteQuery` and `getNextPageParam` here. From the looks of the UX, our pagination is not infinite. If it's not smth obvious that I'm missing, it might be worth documenting this workaround

Reply:
We're using `useInfiniteQuery` in this case because the DataTable allows the user to fetch more data (if available and up to 10k), on the last page:

https://github.com/elastic/kibana/assets/19270322/9d109055-9f1c-4dce-8ee4-79c2556ffcf9

Once all pages are loaded or the max limit (10k records) is reached it no longer shows the load more button (as ex the findings page once user hits loads more until retrieve all records):

<img width="1499" alt="image" src="https://github.com/elastic/kibana/assets/19270322/7e7167ea-a940-4654-804e-a7f18822cd12">



This behavior, even if we're not implementing with scrolling, is similar to infinite scrolling except that requires a "click" to load more data, and the `useInfiniteQuery` differs from `useQuery` because it can accumulate and cache a chunk of data from the previous fetching into an array, and that's what we are using to fetch another chunk of 500 records and accumulating with the previous records, the `getNextPageParam` is useful to know if there are more pages to load and retrieve the position from the last loaded record to be used as a `from` parameter to fetch the next chunk of data.

I believe the confusion here is that the `pages` object and the `getNextPageParam` function are not related to the pages present in the DataTables (which are handled internally on memory by the UnifiedDataTable component), they only relate to the chunk of data that was fetched. I will add a comment on the methods to clarify that usage

Reply:
Thanks for the elaborate answer! I wasn't aware of the load more feature, I thought we just limited the number of findings to the maximum number, good to know!